portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,44 +30,45 @@
 import javax.servlet.http.HttpServletResponse;
 import com.google.sps.servlets.Comment;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that adds a comment to the datastore. */
 @WebServlet("/contact-me")
 public class DataServlet extends HttpServlet {
 
-    @Override
-    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        // Get the input from the form.
-        String name = request.getParameter("name-input");
-        String email = request.getParameter("email-input");
-        String subject = request.getParameter("subject-input");
-        String message = request.getParameter("message-input");
-        long timestamp = System.currentTimeMillis();
-        double sentimentScore = getSentimentScore(message);
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = request.getParameter("name-input");
+    String email = request.getParameter("email-input");
+    String subject = request.getParameter("subject-input");
+    String message = request.getParameter("message-input");
+    long timestamp = System.currentTimeMillis();
 
-        // Error handling - don't allow empty or null values
-        if (!Strings.isNullOrEmpty(name) && !Strings.isNullOrEmpty(email) && !Strings.isNullOrEmpty(subject) && !Strings.isNullOrEmpty(message)) {
-            Entity commentEntity = new Entity("Comment");
-            commentEntity.setProperty("name", name);
-            commentEntity.setProperty("email", email);
-            commentEntity.setProperty("subject", subject);
-            commentEntity.setProperty("message", message);
-            commentEntity.setProperty("timestamp", timestamp);
-            commentEntity.setProperty("sentimentScore", sentimentScore);
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    double sentimentScore = getSentimentScore(message, languageService);
+    languageService.close();
```

Comment:
Sorry, Emily. For some reason, I thought that getSentimentScore was being called multiple times. Since this is not the case I would move the LanguageServiceClient.create()  and close() calls back into getSentimentScore()
```
double getSentimentScore(String message, LanguageServiceClient languageService) throws IOException {
  LanguageServiceClient languageService = LanguageServiceClient.create();
  ...
  languageService.close();
}
```